saleor/payment/gateways/authorize_net/__init__.py

```diff
@@ -0,0 +1,105 @@
+
+    customer_address = apicontractsv1.customerAddressType()
+    customer_address.firstName = payment_information.billing.first_name
+    customer_address.lastName = payment_information.billing.last_name
+    customer_address.company = payment_information.billing.company_name
+    # authorize.net support says we should not attempt submitting street_address_2
+    customer_address.address = payment_information.billing.street_address_1
+    customer_address.city = payment_information.billing.city
+    customer_address.state = payment_information.billing.country_area
+    customer_address.zip = payment_information.billing.postal_code
+    customer_address.country = payment_information.billing.country
+
+    customer_data = apicontractsv1.customerDataType()
+    customer_data.type = "individual"
+    customer_data.id = payment_information.customer_id
+    customer_data.email = payment_information.customer_email
+
+    transaction_request = apicontractsv1.transactionRequestType()
+    transaction_request.transactionType = "authCaptureTransaction"
+    transaction_request.amount = payment_information.amount
+    transaction_request.order = order
+    transaction_request.payment = payment_one
+    transaction_request.billTo = customer_address
+    transaction_request.customer = customer_data
+
+    create_transaction_request = apicontractsv1.createTransactionRequest()
+    create_transaction_request.merchantAuthentication = merchant_auth
+    create_transaction_request.refId = str(payment_information.payment_id)
+    create_transaction_request.transactionRequest = transaction_request
+
+    create_transaction_controller = createTransactionController(
+        create_transaction_request
+    )
+    create_transaction_controller.execute()
+
+    response = create_transaction_controller.getresponse()
+
+    success = False
+    error = None
+    transaction_id = None
+    raw_response = etree.tostring(response).decode()
+    if response is not None:
+        if hasattr(response, "transactionResponse") and hasattr(
+            response.transactionResponse, "transId"
+        ):
+            transaction_id = response.transactionResponse.transId
+        if response.messages.resultCode == "Ok":
+            if hasattr(response.transactionResponse, "messages"):
+                success = True
+            else:
+                if hasattr(response.transactionResponse, "errors"):
+                    error = response.transactionResponse.errors.error[0].errorText
+        else:
+            if hasattr(response, "transactionResponse") and hasattr(
+                response.transactionResponse, "errors"
+            ):
+                error = response.transactionResponse.errors.error[0].errorText
+            else:
+                error = response.messages.message[0]["text"].text
+    else:
+        error = "Null Response"
+
+    if not transaction_id:
+        transaction_id = payment_information.token
+    return GatewayResponse(
```

Comment:
Not sure if authorize.net gives such a possibility, but the `GatewayResponse` has `searchable_key` field which can be used to search assigned order from the API side. 
Example based on the Adyen implementation:
- adyen returns the payment reference
- saleor saves it as a searchable_key
- staff user takes payment reference from Adyen admin panel
- staff user copies the adyen reference to Dashboard/orders search bar
- saleor return orders related to the  provided key.

Reply:
Auhth.net returns a "transId" which is a unique integer only on success. I can set that to both "transaction_id" and "searchable_key" which seems to match how the Adyen plugin does it.

On failure, there's nothing unique I can use.